### PR TITLE
Redesign of events page

### LINKED ISSUE #494: Re-work our main events page
# Proposal
Our current [events page](https://pythondiscord.com/events/) could be improved to draw in more engagement, clearly showcase the upcoming events, and have a stronger call to action for people to join and find the information they're looking for. 

The below image illustrates what I'd like to do.
EventPage MockUp
- A clear intro blurb briefly describing our events and providing a link to go to the server
- A center showcase of our upcoming events with a strong call-to-action to either join or find out more information
- A nice grid layout of our events with buttons that will either "Show More" info that will pop out a modal that provides more information or a "View Event Page" button that will bring you to a dedicated event page (mostly to be used for the code jams). 
- A nice ToC on the side to give an overview of what there is on the page

These are only ideas though and I'm more than happy with the final design looking different as long as it achieves the following objectives:
- Clear information that can be found via skimming
- Upcoming events are easily found with links for how to join / find out more info
- It's easy to see what events we host and generally when they occur

### PR DESCRIPTION
**Pull request by hedyhli**
Closes #494 

This PR attempts address the proposal from the linked issue by redesigning our events page to draw more engagement. This includes making the content more easily accessible and organized.

It is also an opportunity to update the page to be more up-to-date on the latest events for this new year.

Pages on preview:
https://deploy-preview-1210--pydis-static.netlify.app/events/

---

Behold, responsive masonry in CSS only

https://github.com/python-discord/site/assets/50042066/583bc66d-5741-43a6-ad1b-e8dcda1d9f32

---

## Implications

The events page would need to be updated more frequently due to this redesign. This means every time an event starts or finishes, we should update the page.

### Masonry grids

To have a good-looking grid with items of variable heights for our regular and one-off events, masonry was required.

<details><summary>Why do we want Masonry??</summary>
<blockquote>

It isn't easy to have all our event descriptions to have the same visual height.

image

It might be possible to have flexible thumbnail heights though, to compensate the offsets created from the variable description heights. However, to have the Bulma card component sections work with flexbox is not very easy and I found Masonry to be more flexible.

</blockquote>
</details>

For maximum end-user-experience and developer experience, [Masonry itself](https://masonry.desandro.com/) would be probably be the best bet. However I was against adding a whole javascript dependency just for these two sections so I made use of the [newly discovered CSS-only flexbox solution](https://tobiasahlin.com/blog/masonry-with-css/) instead.

We can't simply use Bulma columns because we need to preserve item ordering on mobile, where no columns are used.

<details><summary>What do you mean??</summary>
<blockquote>

If we use bulma columns, we will define a div with class "column" for each visual column. On wide viewports it will show like this:

image

But on mobile, where we disable side-by-side columns it will show like this:

image

<small><a href="https://excalidraw.com/#json=13yf2u-vTLnqdLU9blJpy,b77kg5brRAyKPVbeftMyAA">🔗 Excalidraw</a></small> 

This is not desirable for the regular events grid because we might want to have a specific ordering for the showcase, nor the one-off events because we might want it to be in reverse-chronological order.

</blockquote>
</details>

This came with the limitation that whenever items in the grids are updated we have to modify the `height` in CSS manually, ensuring it has enough room for all screen widths from around 600px to 1000px. This was a pain during development, and for the future when we are adding/removing items from the grid the person who updates it should not be overly involved in the CSS.

The implementation is in the `events/base.css` CSS file with comments.

## Ideas for the future

- #1269 

## Assets

Should use the same icons for the same event, or should we use the annual branding? For instance, we have different branding for each Pyweek and Code Jam, should the icon used in the "scheduled"/"previous" events use the specific branding for that year, this makes multiple entries of the same event have different icons.

Currently, I use the same Pyweek icons, but for Code Jam 2023, I used the 2023 icon, and for the rest, used 2022 icon.

## Todo

Design
- [x] Implement masonry layout, hopefully without JS dependency
- [x] Consider using more fancy colors than white
- [x] Fix up spacing
- [x] Fix layout on tablets
- [x] Ensure consistency on mobile
- [x] Reconsider better labels and title text

Content
- [x] Add one-off events list
- [x] Apply changes suggested by Kat (on owners qna)
- ~~Finalize the assets to use~~
- [x] Finalize list and order of events
- [x] Ensure event schedules are correct
- [x] Fill up all the event pages

Etc.
- [x] Do dark theme stuff

--- Optimistic deadline: February ---
--- WIP ➡️ ready for reviews ---

Maybe
- [x] Minify and/or lazy-load all the images
- [x] Document in a readme how to update info in the future for new events
- Should the events page get a special dazzling hero section similar to the home page with a confetti icon?

### CODE REVIEW

**Comment by hedyhli** [comment]
There are still a number of issues with the design, but I've run out of ideas on how to best solve them. So I'll mark this PR as ready for review now.

**Review by janine9vn — APPROVED** [review]
I didn't take an in-depth look at the code because I more-so care how it presents (and html is only so interesting). Overall, I love it! It presents what we do really well and is really and easily digestible.

Thank you so much for your work on this.

File: pydis_site/templates/events/other_events.html [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            <div class="media-content">
+                <div class="content">
+                    <h2 class="title is-4"><a href="https://www.youtube.com/watch?v=PGZPSWZSkJI">Python 3.11 Release Stream</a></h2>
+                    <p class="subtitle is-7 has-text-grey">
+                        <span class="icon-text">
+                            <span class="date-icon icon"><i class="fa-regular fa-calendar"></i></span>
+                            <span>Oct. 25, 2022</span>
+                        </span>
+                    </p>
+                    <p>Pablo Galindo, the 3.11 Python Release Manager and CPython Core Developer, walked through the release process and g number of guests joined to talk about the exciting new features to be found in 3.11. At the end of the release we'll also hosted a Q&A session.</p>
```
Comment (jchristgit) [inline pydis_site/templates/events/other_events.html]:
```suggestion
                    <p>Pablo Galindo, the 3.11 Python Release Manager and CPython Core Developer, walked through the release process and a number of guests joined to talk about the exciting new features to be found in 3.11. At the end of the release we also hosted a Q&A session.</p>
```

File: pydis_site/templates/events/pages/game-jams/_index.html [hunk 2]
```diff
@@ -8,5 +8,25 @@
...
+    <div class="column">
+        <p>
+            The Game Jam is similar to our Code Jams, but smaller in scope. Instead of having to complete a qualifier
+            and being teamed up with random strangers, members of our community can just sign-up individually or pair up
+            with whoever they like.
+        </p>
+        <p>
+            The participants will have ten days to create a game using the technology we've selected, and drawing
+            inspiration from a provided theme. After the event, a panel of judges will play all the games and select a
+            winner. The top 5 will featured in a special video on our <a href="https://www.youtube.com/channel/UCQsrA4xo6jvdgsJZhKaBL6w" target="_blank" rel="noopener">YouTube channel</a>.
```
Comment (jchristgit) [inline pydis_site/templates/events/pages/game-jams/_index.html]:
```suggestion
            winner. The top 5 will be featured in a special video on our <a href="https://www.youtube.com/channel/UCQsrA4xo6jvdgsJZhKaBL6w" target="_blank" rel="noopener">YouTube channel</a>.
```

File: pydis_site/templates/events/pages/pyweek/_index.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <li><a href="{% url "events:index" %}">Events</a></li>
+    <li class="is-active"><a href="#">PyWeek</a></li>
+{% endblock %}
+
+{% block title %}PyWeek{% endblock %}
+
+{% block event_content %}
+<p>
+    For the past 15 years, <a href="https://pyweek.org" target="_blank" rel="noopener">PyWeek</a> has been running a bi-annual game jam for the
+    Python language. As of 2020, we are excited to say we are officially partnered with PyWeek to co-run these
```
Comment (jchristgit) [inline pydis_site/templates/events/pages/pyweek/_index.html]: Hmm, I'm not sure if this is necessary from a grammatical perspective, but I feel it reads more smoothly this way:
```suggestion
    Python language. As of 2020, we are excited to say that we are officially partnered with PyWeek to co-run these
```
What do you think?

**Review by jchristgit — CHANGES_REQUESTED** [review]
Looks very good to me, thank you for the hard work!

I have a few tiny nitpicks, after that it looks good to merge to me. Thank you again!

File: pydis_site/templates/events/pages/pyweek/_index.html (hunk 3 above)
Comment (hedyhli) [reply pydis_site/templates/events/pages/pyweek/_index.html]: This was copied from the current description on our events page, but I agree! It's a good suggestion -- it sounds better.

File: pydis_site/templates/events/pages/game-jams/_index.html (hunk 2 above)
Comment (hedyhli) [reply pydis_site/templates/events/pages/game-jams/_index.html]: This too, our current page sure has some grammatical errors. Thanks!

File: pydis_site/templates/events/pages/pyweek/_index.html (hunk 3 above)
Comment (hedyhli) [reply pydis_site/templates/events/pages/pyweek/_index.html]: ece9c315

Also fixed another instance of this content on the annual events gallery.

**Review by jchristgit — APPROVED** [review]

**Pull request merged**